Web/Scotty/Internal/Types.hs

```diff
@@ -156,24 +162,46 @@ data ScottyException
   | QueryParameterNotFound Text
   | FormFieldNotFound Text
   | FailedToParseParameter Text Text Text
+  | WarpRequestException W.InvalidRequest
+  | WaiRequestParseException WPS.RequestParseException -- request parsing
+  | ResourceTException RT.InvalidAccess -- use after free
   deriving (Show, Typeable)
 instance E.Exception ScottyException
 
 ------------------ Scotty Actions -------------------
 type Param = (Text, Text)
 
-type File = (Text, FileInfo LBS8.ByteString)
+-- | Type parameter @t@ is the file content. Could be @()@ when not needed or a @FilePath@ for temp files instead.
+type File t = (Text, FileInfo t)
 
-data ActionEnv = Env { envReq       :: Request
+data ActionEnv = Env { envInternalState :: RT.InternalState
+                     , envParseRequestBodyOpts :: WPS.ParseRequestBodyOptions
```

Comment:
`envInternalState` and `envParseRequestBodyOpts` are only needed by the current implementation of `files`.

Reply:
fixed